test/units/modules/test_iptables.py

```diff
@@ -1,1174 +1,1386 @@
+    )
+
+    with pytest.raises(SystemExit):
+        iptables.main()
+
+    assert run_command.call_count == 1
+    first_cmd_args_list = run_command.call_args_list[0]
+    assert first_cmd_args_list[0][0] == [
+        IPTABLES_CMD,
+        "-t",
+        "filter",
+        "-C",
+        "OUTPUT",
+        "-w",
+        "10",
+        "-s",
+        "1.2.3.4/32",
+        "-d",
+        "7.8.9.10/42",
+        "-j",
+        "ACCEPT",
+    ]
+
+
+def test_comment_position_at_end(mocker, mock_basic_commands):
+    """Test comment position to make sure it is at the end of command."""
+    set_module_args(
+        {
+            "chain": "INPUT",
+            "jump": "ACCEPT",
+            "action": "insert",
+            "ctstate": ["NEW"],
+            "comment": "this is a comment",
+            "_ansible_check_mode": True,
+        }
+    )
+
+    commands_results = [
+        (0, "", ""),
+    ]
+
+    run_command = mocker.patch(
+        "ansible.module_utils.basic.AnsibleModule.run_command",
+        side_effect=commands_results,
+    )
+
+    with pytest.raises(SystemExit):
+        iptables.main()
+
+    assert run_command.call_count == 1
+    first_cmd_args_list = run_command.call_args_list[0]
+    assert first_cmd_args_list[0][0] == [
+        IPTABLES_CMD,
+        "-t",
+        "filter",
+        "-C",
+        "INPUT",
+        "-j",
+        "ACCEPT",
+        "-m",
+        "conntrack",
+        "--ctstate",
+        "NEW",
+        "-m",
+        "comment",
+        "--comment",
+        "this is a comment",
+    ]
+    assert run_command.call_args[0][0][14] == "this is a comment"
+
+
+def test_destination_ports(mocker, mock_basic_commands):
+    """Test multiport module usage with multiple ports."""
+    set_module_args(
+        {
+            "chain": "INPUT",
+            "protocol": "tcp",
+            "in_interface": "eth0",
+            "source": "192.168.0.1/32",
+            "destination_ports": ["80", "443", "8081:8085"],
+            "jump": "ACCEPT",
+            "comment": "this is a comment",
+        }
+    )
+    commands_results = [
+        (0, "", ""),
+    ]
+    run_command = mocker.patch(
+        "ansible.module_utils.basic.AnsibleModule.run_command",
+        side_effect=commands_results,
+    )
+
+    with pytest.raises(SystemExit):
+        iptables.main()
+
+    assert run_command.call_count == 1
+    first_cmd_args_list = run_command.call_args_list[0]
+    assert first_cmd_args_list[0][0] == [
+        IPTABLES_CMD,
+        "-t",
+        "filter",
+        "-C",
+        "INPUT",
+        "-p",
+        "tcp",
+        "-s",
+        "192.168.0.1/32",
+        "-j",
+        "ACCEPT",
+        "-m",
+        "multiport",
+        "--dports",
+        "80,443,8081:8085",
+        "-i",
+        "eth0",
+        "-m",
+        "comment",
+        "--comment",
+        "this is a comment",
+    ]
+
+
+@pytest.mark.parametrize(
+    "test_input,expected",
+    [
+        (
```

Comment:
Such complex data structures produce hard to understand test IDs. I recommend coming up with IDs manually. For this, wrap each param into an explicit `pytest.param` instance and pass `id='custom-param-explanation'` to them.
```suggestion
        pytest.param(
```